jobserver/api/releases.py

```diff
@@ -259,6 +303,48 @@ def get(self, request, file_id):
         return serve_file(request, rfile)
 
 
+class ReviewAPI(APIView):
+    authentication_classes = [SessionAuthentication]
+
+    class ReleaseSerializer(serializers.Serializer):
+        files = FileSerializer(many=True)
+        metadata = serializers.DictField()
+        review = serializers.DictField()
+
+    def post(self, request, *args, **kwargs):
+        release = get_object_or_404(Release, id=self.kwargs["release_id"])
+        db_files = release.files.all()
+
+        serializer = self.ReleaseSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        payload_files = serializer.validated_data["files"]
+
+        with transaction.atomic():
+            release.metadata = serializer.validated_data["metadata"]
+            release.review = serializer.validated_data["review"]
+            release.save(update_fields=["metadata", "review"])
+
+            try:
+                validate_files(db_files, payload_files)
+            except UnknownFiles as e:
+                sentry_sdk.capture_exception(e)
+                raise ValidationError(e)
+
+            # create review objects for each file in the files list
+            for file in payload_files:
+                rfile = db_files.get(name=file["name"], filehash=file["sha256"])
+
+                created_at = timezone.now()
```

Comment:
Move this timezone.now() to be outside the loop, so it's the same for all ReleaseFileReview's created by this review?

Reply:
Yep 🤦‍♂️